app/services/vacancies/import/sources/vacancy_poster.rb

```diff
@@ -0,0 +1,128 @@
+require "cgi"
+
+class Vacancies::Import::Sources::VacancyPoster
+  FEED_URL = ENV.fetch("VACANCY_SOURCE_VACANCY_POSTER_FEED_URL").freeze
+  VENTRUS_TRUST_UID = "".freeze
+  SOURCE_NAME = "vacancy_poster".freeze
+
+  class FeedItem
+    def initialize(xml_node)
+      @xml_node = xml_node
+    end
+
+    def [](key)
+      @xml_node.xpath(key)&.text&.presence
+    end
+  end
+
+  include Enumerable
+
+  def self.source_name
+    SOURCE_NAME
+  end
+
+  def each
+    items.each do |item|
+      v = Vacancy.find_or_initialize_by(
+        external_source: SOURCE_NAME,
+        external_reference: item["reference"]&.strip,
+      )
+
+      # An external vacancy is by definition always published
+      v.status = :published
+      # Consider publish_on date to be the first time we saw this vacancy come through
+      # (i.e. today, unless it already has a publish on date set)
+      v.publish_on ||= Date.today
+
+      begin
+        v.assign_attributes(attributes_for(item))
+      rescue ArgumentError => e
+        v.errors.add(:base, e)
+      end
+
+      yield v
+    end
+  end
+
+  def attributes_for(item)
+    {
+      job_title: item["jobTitle"],
+      job_advert: job_advert_for(item),
+      salary: item["salary"],
+      expires_at: Time.zone.parse(item["expiresAt"]),
+      external_advert_url: item["advertUrl"]&.strip,
+
+      job_roles: job_roles_for(item),
+      ect_status: ect_status_for(item),
+      key_stages: item["keyStages"].presence&.split(","),
+      subjects: item["subjects"].presence&.split(","),
+      working_patterns: item["workingPatterns"].presence&.split(","),
+      contract_type: item["contractType"].presence,
+      phases: phase_for(item),
+      visa_sponsorship_available: false,
+    }.merge(organisation_fields(item))
+  end
+
+  def job_advert_for(item)
+    sanitised_text = Rails::Html::WhiteListSanitizer.new.sanitize(item["jobAdvert"], tags: %w[p br])
+    CGI.unescapeHTML(sanitised_text)
+  end
+
+  def ect_status_for(item)
+    return unless item["ectSuitable"].presence
+
+    item["ectSuitable"] == "yes" ? "ect_suitable" : "ect_unsuitable"
+  end
+
+  def organisation_fields(item)
+    schools = find_schools(item)
+    first_school = schools.first
+
+    {
+      organisations: schools,
+      readable_job_location: first_school&.name,
+      about_school: first_school&.description,
+    }
+  end
+
+  def find_schools(item)
+    multi_academy_trust = SchoolGroup.trusts.find_by(uid: item["trustUID"]&.strip)
+
+    multi_academy_trust&.schools&.where(urn: item["schoolUrns"]&.strip).presence ||
+      Organisation.where(urn: item["schoolUrns"]).presence ||
+      Array(multi_academy_trust)
+  end
```

Comment:
There is a new implementation for "find_schools" that has been applied to multiple sources. 
It allows MAT central office vacancies and fixes some edge case issues in this initial code. It is being implemented across the sources (and we plan to extract it to become a shared code helper for all the sources):
https://github.com/DFE-Digital/teaching-vacancies/blob/main/app/services/vacancies/import/sources/ark.rb#L182-L192